pkg/keymanagementprovider/azurekeyvault/provider.go

```diff
@@ -59,6 +62,7 @@ type AKVKeyManagementProviderConfig struct {
 	TenantID     string                `json:"tenantID"`
 	ClientID     string                `json:"clientID"`
 	CloudName    string                `json:"cloudName,omitempty"`
+	Resource     string                `json:"resource,omitempty"`
```

Comment:
Is this value exposed to the user for configuration? What happens when they set a value to `resource` field? I see that this field is set in the controller. But I'm wondering why we need to JSON serialize this field if it's not exposed as a configurable user field?

Reply:
Good question. It should be exposed to the user for configuration. The controller is responsible for populating it. So, the json tag can be removed. I'll get that changed.

Reply:
QQ: What about other fields like `Type`, and `CloudName`?

Reply:
Good point. The value doesn't need to be exposed. And there's no reason to JSON serialize it. 

If the `resource` isn't passed into the create method via the `kmProviderConfig`, where would you suggesting passing it in? Perhaps, passing it into `factory.CreateKeyManagementProviderFromConfig` would work.

Reply:
Ok I see the issue. The config struct is currently the only place to pass the `resource` to the AKV KMP in the `Create` method. And we need to JSON serialize since the whole config is Marshalled and then Unmarshalled into the type.

Reply:
>Perhaps, passing it into factory.CreateKeyManagementProviderFromConfig would work.

This would in turn require updating the KMP factory API 'Create' to incorporate another parameter for `resource`. IN the future if we need to pass even more non config info to the each KMP, adding parameters to the API is not extensible. I think this is a larger discussion not just specific to this PR. For now, let's just proceed as is.